zebrad/tests/common/lightwalletd/send_transaction_test.rs

```diff
@@ -296,25 +288,6 @@ async fn send_transactions_from_block(
         .await?
         .into_inner();
 
-    // Sometimes lightwalletd doesn't check the mempool, and waits for the next block instead.
-    // If that happens, we skip the rest of the test.
-    tracing::info!("checking if lightwalletd has queried the mempool...");
-
-    // We need a short timeout here, because sometimes this message is not logged.
-    zebrad = zebrad.with_timeout(Duration::from_secs(60));
-    let tx_log =
-        zebrad.expect_stdout_line_matches("answered mempool request .*req.*=.*TransactionIds");
-    // Reset the failed timeout and give the rest of the test enough time to finish.
-    #[allow(unused_assignments)]
-    {
-        zebrad = zebrad.with_timeout(LARGE_CHECKPOINT_TIMEOUT);
-    }
-
-    if tx_log.is_err() {
-        tracing::info!("lightwalletd didn't query the mempool, skipping mempool contents checks");
-        return Ok((zebrad, has_tx_with_shielded_elements, 0));
-    }
-
```

Comment:
It looks like lightwalletd is [always querying `getrawmempool` as long as it's been 2s since the last query](https://github.com/zcash/lightwalletd/blob/master/frontend/service.go#L471-L475), and the body of the two loops below shouldn't execute if that ever changes.

There might be another problem here, it's checking that the ids of transactions it gets from lightwalletd are all in `transaction_hashes`, but lightwalletd may return transactions that were added to the mempool from previous blocks, and `transaction_hashes` only includes ids of transactions in the current block. I'll fix that.